server/src/pages/shotindex/controller.js

```diff
@@ -173,6 +173,46 @@ exports.deleteShot = function(shot) {
   req.send(`id=${encodeURIComponent(shot.id)}&_csrf=${encodeURIComponent(model.csrfToken)}`);
 };
 
+
+exports.toggleFavoriteShot = async function(shot) {
+  const url = model.backend + "/api/set-expiration";
+  const newExpiration = !shot.expireTime ? model.defaultExpirationMs : 0;
+  const body = new URLSearchParams();
+
+  body.append("id", shot.id);
+  body.append("expiration", newExpiration);
+  body.append("_csrf", model.csrfToken);
+
+  try {
+    const resp = await fetch(url, {
+      method: "POST",
+      body,
+    });
+
+    if (!resp.ok) {
+      window.Raven.captureException(
+        new Error(`Error calling /api/set-expiration: ${resp.status} ${resp.statusText}`));
+      throw new Error();
+    }
+
+    if (newExpiration) {
+      shot.expireTime = Date.now() + newExpiration;
+    } else {
+      shot.expireTime = null;
+    }
+
+    render();
+  } catch (err) {
+    // The error could be one from the server or a network level error from
+    // fetch; we display a generic message either way as it's (probably?)
+    // friendlier than percolating a fetch error message up to the user.
```

Comment:
Probably good to do `console.warn("Error updating expiration:", err);` so it isn't entirely swallowed